settings.c

```diff
@@ -268,9 +268,50 @@ static void UaServer_FSBE_RemoveEntry(FileSettings *pFS, int index)
         free(pEntry->pszValue);
         pEntry->pszValue = 0;
     }
-    pEntry->type = Invalid;
-    pEntry->parent = -1;
-    pEntry->line = 0;
+
```

Comment:
I'd  an assert here for numEntries > 0

Reply:
What should I use as "assert"? Currently there is no assert in any code...

Reply:
numEntries > 0 is wrong in this situation... we already have decremented the numEntries, so we must check for "numEntries >= 0"; we must reset the now cleared entry with the default values...